uc-crux-llvm/src/UCCrux/LLVM/Overrides/Skip.hs

```diff
@@ -145,10 +166,196 @@ unsoundSkipOverrides modCtx sym mtrans usedRef annotationRef postconditions decl
+  \case
+    ClobberSelectArgument idx cursor -> SelectArgument idx cursor
+    ClobberSelectGlobal glob cursor -> SelectGlobal glob cursor
+
+clobberSelectorCursor ::
+  ClobberSelector m argTypes inTy atTy ->
+  Cursor m inTy atTy
+clobberSelectorCursor =
+  \case
+    ClobberSelectArgument _idx cursor -> cursor
+    ClobberSelectGlobal _glob cursor -> cursor
+
+-- | What data should this override clobber?
+--
+-- These specs can explicitly set a type of data to generate (@argTypes@,
+-- @inTy@, @atTy@), which may or may not actually match the declared type of the
+-- data in question. This is useful for clobbering e.g. a @char*@ or @void*@
+-- with structured data. However, it yields some API and implementation
+-- complexity due to possible type mismatches (see 'ClobberSpecError').
+data ClobberSpec m (argTypes :: Ctx (FullType m)) inTy atTy
+  = ClobberSpec
+      { clobberSelector :: ClobberSelector m argTypes inTy ('FTPtr atTy),
+        -- | Type of value that contains the pointer
+        clobberType :: FullTypeRepr m inTy,
+        -- | Constraints on data to write
+        clobberShape :: ConstrainedShape m atTy
+      }
+  deriving Eq
+
+data SomeClobberSpec m
+  = forall (argTypes :: Ctx (FullType m)) inTy atTy.
+      SomeClobberSpec (ClobberSpec m argTypes inTy atTy)
+
+data ClobberSpecError m
+  = ClobberMismatchedSize !(FuncSymbol m)
+  | ClobberMismatchedType !(FuncSymbol m)
+  deriving (Eq, Ord)
+
+ppClobberSpecError :: ClobberSpecError m -> PP.Doc ann
+ppClobberSpecError =
+  \case
+    ClobberMismatchedSize f ->
+      PP.hsep
+        [ "Specification for values clobbered by the skipped function",
+          PP.pretty (funcSymbolToString f),
+          "included an argument index that is out of range for that function."
+        ]
+    ClobberMismatchedType f ->
+      PP.hsep
+        [ "Specification for values clobbered by the skipped function",
+          PP.pretty (funcSymbolToString f),
+          "included a value that was ill-typed with respect to that function's",
+          "arguments."
+        ]
+
+-- | Retrieve the value specified by a 'ClobberSelector'.
+--
+-- 'makeGetter' returns this sort of callback because we want to produce
+-- 'ClobberSpecError's statically, that is, before executing the override, and
+-- this is a relatively simple way to encapsulate the type information learned
+-- by GADT case splits in 'makeGetter'.
+newtype ClobberGetter arch args inTy =
+  ClobberGetter
+    { getClobberGetter ::
+        forall sym.
+        MemImpl sym ->
+        Ctx.Assignment (Crucible.RegEntry sym) args ->
+        IO (Crucible.RegValue sym (ToCrucibleType arch inTy))
+    }
+
+-- | Retrieve a getter for the value with the pointer to be clobbered.
+--
+-- See comment on 'ClobberGetter'.
+makeGetter ::
+  forall proxy m arch args argTypes inTy atTy.
+  ArchOk arch =>
+  proxy arch ->
+  FuncSymbol m ->
+  Ctx.Assignment CrucibleTypes.TypeRepr args ->
+  ClobberSpec m argTypes inTy atTy ->
```

Comment:
That's all exactly right. This is explicitly designed so that you can make UC-Crux produce a `my_struct*` where it sees a `i8*` (i.e. `char*`/`void*`) in the function's declared signature.